service/internal/graph/graph.go

```diff
@@ -578,3 +580,161 @@ func connectorStability(f connector.Factory, expType, recType component.Type) co
 	}
 	return component.StabilityLevelUndefined
 }
+
+// TODO: remove as part of https://github.com/open-telemetry/opentelemetry-collector/issues/7370 for service 1.0
+type getExporters interface {
+	GetExporters() map[component.DataType]map[component.ID]component.Component
+}
+
+var _ getExporters = (*Host)(nil)
+var _ component.Host = (*Host)(nil)
+
+type Host struct {
```

Comment:
I moved `host.go` from the root of the component dir to here bc I needed `graph` to be specific to our host instead of using `component.Host`.

I wanted access to our specific host so that I could access its `service/internal/status.Reporter` instance, which has some additional methods, like `ReportOKIfStarting` that the graph needs access to that `componentstatus.Reporter` does not have.

I could experiment with checking if the `component.Host` implements `service/internal/status.Reporter` instead of passing in our host directly.  That will reduce the number of changes, but require some extra interface checks in the graph's `StartAll`.

Reply:
It seems odd to define it here long term, but I don't think it necessarily needs to be reworked immediately.

Reply:
I agree that it is odd.  Luckily it is all unexported/internal stuff so I agree that it could be reworked another time.